uPortal-layout/uPortal-layout-core/src/main/java/org/apereo/portal/layout/IUserLayoutStore.java

```diff
@@ -40,6 +40,21 @@ public interface IUserLayoutStore {
      */
     DistributedUserLayout getUserLayout(IPerson Person, IUserProfile profile);
 
+    /**
+     * Provides a {@link Tuple} containing the &quot;fragmentized&quot; version of a DLM fragment
+     * owner's layout, together with the username. This version of the layout consistent with what
+     * DLM uses internally for fragments, and is created by FragmentActivator.fragmentizeLayout.
+     * It's important that the version returned by this method matches what DLM uses internally
+     * because it will be used to establish relationships between fragment layout nodes and user
+     * customizations of DLM fragments.
+     *
+     * @param userName The username of the user for whom the layout is retrieved.
+     * @param userId The unique identifier of the user.
+     * @return A {@link Tuple} containing the username and the "fragmentized" version of the DLM
+     *     fragment owner's layout.
+     */
+    Tuple<String, DistributedUserLayout> getUserLayoutTuple(String userName, int userId);
```

Comment:
The original intent, commented on the original code, was to make the method private.
I'm not sure moving it and making it public (instead of default) makes it clearer.

Reply:
I believe the original method's intent to make this method private was based on the fact that this method is exclusive to this class only. But, this also introduces the feature envy design level smell, which can be addressed by the move method refactor I applied. 
Since this was was anyways solely relying on the other class public methods without using any direct reference from the original class it was in, I believe it made sense to move it to the class where it should belong.

https://luzkan.github.io/smells/feature-envy - based on Martin Fowler's Paper

https://swiftscribe.medium.com/code-smells-couplers-feature-envy-c54efaf5eef3

Reply:
I'm well aware of uncle bob and his adages. 🙂 
They can be useful rules of thumb, but are not necessarily gospel.

Other @uPortal-Project/uportal-committers thoughts on this?

Reply:
I would be more willing to accept this refactor if it was used elsewhere but there has not been any _outside_ uses to justify to me that this needs to be moved from where it is at the moment.
Do not get me wrong: refactoring is good from time to time, but over-refactoring could have the opposite desired effects.

Reply:
Thanks @ChristianMurphy and @jonathanmtran for giving your valuable inputs and feedbacks.
There is a lot to learn from your perspectives. 

@ChristianMurphy, 
Is everything okay to be merged aside form this move refactor commit ? If yes, I will create a separate PR for all the acceptable commit changes and get it merged. This would clean up the further action items from my end.